test/e2e/queue.go

```diff
@@ -581,13 +581,24 @@ var _ = Describe("AppWrapper E2E Test", func() {
 		aw := createGenericJobAWWithStatus(context, "aw-test-job-with-comp-1")
 		err1 := waitAWPodsReady(context, aw)
 		Expect(err1).NotTo(HaveOccurred())
-		time.Sleep(1 * time.Minute)
 		aw1, err := context.karclient.McadV1beta1().AppWrappers(aw.Namespace).Get(context.ctx, aw.Name, metav1.GetOptions{})
 		if err != nil {
 			fmt.Fprintf(GinkgoWriter, "Error getting status, %v\n", err)
 		}
 		Expect(err).Should(Succeed())
-		Expect(aw1.Status.State).To(Equal(arbv1.AppWrapperStateCompleted))
+		var pass = false
+		for true {
```

Comment:
Can we adopt a more functional style. using idiomatic Gomega, and avoid that `for true` loop, e.g. with:

```go
Eventually(AppWrapper(test, namespace, aw.Name)).
	Should(WithTransform(AppWrapperState, Equal(mcadv1beta1.AppWrapperStateActive)))
```

There is an example in the CodeFlare operator e2e tests:

https://github.com/project-codeflare/codeflare-operator/blob/be0dd48b43b415e8fe06bdf0c940d53c088619b3/test/e2e/mnist_rayjob_mcad_raycluster_test.go#L225-L226